File: 1/go/part1.go
package main

import (
	"bufio"
	"io"
	"log"
	"math"
	"os"
	"sort"
	"strconv"
	"strings"
)

func part1(filename string) {

	file, err := os.Open(filename)
	if err != nil {
		log.Fatal(err)
	}
	defer file.Close()

	r := bufio.NewReader(file)

	var list1 []int
	var list2 []int
	var totalValue int

	for {
		line, _, err := r.ReadLine()
		if err == io.EOF {
			break
		}
		if err != nil {
			log.Fatal(err)
		}
		if len(line) > 1 {
			values := strings.Split(string(line), "   ")
			value1, err := strconv.Atoi(values[0])
			if err != nil {
				log.Fatal(err)
			}
			value2, err := strconv.Atoi(values[1])
			if err != nil {
				log.Fatal(err)
			}
			list1 = append(list1, value1)
			list2 = append(list2, value2)
		}
	}

	sort.Ints(list1)
	sort.Ints(list2)

	for i, value := range list1 {
		totalValue += int(math.Abs(float64(value - list2[i])))
	}

	log.Println(totalValue)
}

File: 1/py/part1.py
import sys

inputFile = sys.argv[1]
input = ""

with open(inputFile, "r") as f:
    input = f.read()

list1 = []
list2 = []
total_distance = 0

for row in input.split("\n"):
    splitted = row.split("   ")
    list1.append(int(splitted[0]))
    list2.append(int(splitted[-1]))

list1.sort()
list2.sort()

for i in range(len(list1)):
    total_distance += abs(list1[i]-list2[i])

print(total_distance)